lib/logatron/logatron.rb

```diff
@@ -50,12 +50,27 @@ def debug(msg)
       logger.debug(msg)
     end
 
+    # @deprecated Use operation_context for transportation-layer neutral data.
     def http_headers
       {
-        'X-Ascent-Log-Id' => msg_id
+        'X-Ascent-Log-Id' => msg_id,
+        'X-Ascent-Site' => site
       }
     end
 
+    def operation_context
+      {
+        message_id: msg_id,
+        site: site
+      }
+    end
+
+    def operation_context=(info)
+      return unless info.is_a? Hash
+      self.msg_id = info[:message_id] || info['message_id'] || self.msg_id
+      self.site = info[:site] || info['site'] || self.site
```

Comment:
Question.. what about making this take keyed parameters? That would simplify the following

1. hash class check
2. symbol vs string key check

It would also allow for adding of options without forcing all clients of the method to change at the same time. You could even set the defaults...

It just keeps getting better as I type...

```ruby
def set_operation_context(site: self.site, message_id: self.msg_id)
  self.msg_id = message_id
  self.site = site
end
```